core/domain/platform_parameter_domain.py

```diff
@@ -718,16 +766,17 @@ def to_dict(self):
             'feature_stage': self._feature_stage
         }
 
-    def _validate_feature_flag(self):
+    # We are comparing the same types
+    def _validate_feature_flag(self) -> None:
         """Validates the PlatformParameter domain object that is a feature
         flag.
         """
-        if self._data_type != DATA_TYPES.bool.value:
+        if self._data_type != DATA_TYPES.bool.value: # type: ignore[comparison-overlap]
```

Comment:
Again, we shouldn't use so much ignores, please try to get rid of the or explain them with a comment.

Reply:
done